primitives/src/pcs/mod.rs

```diff
@@ -54,18 +51,28 @@ pub trait PolynomialCommitmentScheme {
     /// Batch proofs
     type BatchProof: Clone + CanonicalSerialize + CanonicalDeserialize + Debug + PartialEq + Eq;
 
-    /// Build SRS for testing.
+    /// Setup for testing.
     ///
     /// - For univariate polynomials, `supported_size` is the maximum degree.
     /// - For multilinear polynomials, `supported_size` is the number of
     ///   variables.
     ///
     /// WARNING: THIS FUNCTION IS FOR TESTING PURPOSE ONLY.
     /// THE OUTPUT SRS SHOULD NOT BE USED IN PRODUCTION.
-    fn gen_srs_for_testing<R: RngCore + CryptoRng>(
+    fn setup_for_testing<R: RngCore + CryptoRng>(
         rng: &mut R,
         supported_size: usize,
-    ) -> Result<Self::SRS, PCSError>;
+    ) -> Result<Self::SRS, PCSError> {
+        Self::SRS::gen_srs_for_testing(rng, supported_size)
+    }
+
+    /// Setup public parameter in production environment.
+    /// These parameters are loaded from files with serialized `pp` bytes, and
+    /// the actual setup is usually carried out via MPC and should be
+    /// implemented else where. We only load them into memory here.
+    fn setup(_supported_size: usize) -> Result<Self::SRS, PCSError> {
```

Comment:
- the name is from a standard PCS def, any PCS should have Setup, Commit, Open, Verify etc. right? Internally, yes, I'm expecting to be calling `load_srs_from_file()` internal functions. But to the external code, generic `setup()` seems like the right abstraction level, wdyt?
- mostly for completeness of a PCS def.
- yes, that's likely what we will do, as we are already loading Aztec's SRS in our [CAP project](https://github.com/EspressoSystems/cap/blob/main/src/parameters.rs#L82).

Reply:
> * the name is from a standard PCS def, any PCS should have Setup, Commit, Open, Verify etc. right? Internally, yes, I'm expecting to be calling `load_srs_from_file()` internal functions. But to the external code, generic `setup()` seems like the right abstraction level, wdyt?

This is a matter of opinion, but I say we don't need setup, and consistency with PCS def is not a good reason to add it when it's not needed. Basically, users should never use their own setup in production code. If we provide a `setup` then that suggests that users should use it. But that's bad practice because any params generated from `setup` cannot be trusted.

Reply:
I actually was thinking inside `setup()` we would load from files. 
or else, are you suggesting that we add `fn load_srs_from_file()` into the trait?

Reply:
Ah sorry I missed that in the comments. I saw only `setup(_supported_size: usize)` and mistakenly concluded that the intention is to actually create a new (untrusted) setup.

Need to think more about the right API.
- *If* we want to provide this functionality at the trait level *then* yes I'd call it something like `load_srs_from_file`.
- We should also provide convenient access to any hard-coded setups we will eventually ship, probably via a different function `load_srs_hard_coded` or whatever.
- Is there any way to move `gen_srs_for_testing` out of production code? We'd have to take it outside the `StructuredReferenceString` trait. Maybe we could move it to a separate `StructuredReferenceStringTest: StructuredReferenceString` trait that's included only in test code.

Reply:
- okay, I will change the trait fn name.
- in eventual production code, we would use `load_srs_from_file(None)` which will load the default/hard-coded binary. 
- yes, we could probably hide the function under `feature="test"` flag.

Reply:
phew... that's a somewhat big change, see [c6a74d4](https://github.com/EspressoSystems/jellyfish/pull/231/commits/c6a74d4d576d8961d7bf649f7d0b43a4f2aba1c6) 

also cc @chancharles92, @mrain since this could be a controversial change.

Reply:
ah... I think this is a mistake on second thought. 🤦‍♂️ 

Other crates (like consensus code) will need to call this `_for_testing` anyway, we should ship this API instead of only conditionally compile it under `cfg(any(test, feature = "bench"))`. 

(also not being able to `exclude` `plonk/examples/proof_of_exp.rs` from `cargo test` is a pain, I tried many suggestions found online, none of them work. This is causing CI failure in the above commit)